egograph/pipelines/api/compaction.py

```
@@ -0,0 +1,83 @@
+@router.get("/datasets")
+def list_datasets(
+    _: None = Depends(verify_api_key),
+    service: PipelineService = Depends(get_service),
+) -> list[dict]:
+    """dataset catalog の一覧を取得する。"""
+    return service.list_datasets()
+
+
+@router.post("/compaction/runs", status_code=201)
+def create_compaction_run(
+    request_body: dict = Body(...),
```

Ruff (0.16.0)

[warning] Do not perform function call `Body` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)